packages/oauth/oauth_tests.js

```diff
@@ -59,3 +59,51 @@ Tinytest.add(
     test.equal(OAuth._retrievePendingCredential(key, secret), cred);
   }
 );
+
+Tinytest.add("oauth - _endOfLoginResponse with popup loginStyle supports ROOT_URL_PATH_PREFIX",
+  function (test) {
+    var rootUrlPathPrefix = __meteor_runtime_config__.ROOT_URL_PATH_PREFIX;
+    __meteor_runtime_config__.ROOT_URL_PATH_PREFIX = '/test-root-url-prefix';
+    var res = {
+      writeHead: function () {},
+      end: function (content) {
+        __meteor_runtime_config__.ROOT_URL_PATH_PREFIX = rootUrlPathPrefix;
+        test.matches(
+          content,
+          /\/test-root-url-prefix\/packages\/oauth\/end_of_popup_response\.js/
+        );
+      }
+    };
+    var details = {
+      credentials: {},
+      loginStyle: 'popup'
+    };
+    OAuth._endOfLoginResponse(res, details);
+  }
+);
+
+Tinytest.add("oauth - _endOfLoginResponse with redirect loginStyle supports ROOT_URL_PATH_PREFIX",
+  function (test) {
+    var rootUrlPathPrefix = __meteor_runtime_config__.ROOT_URL_PATH_PREFIX;
+    __meteor_runtime_config__.ROOT_URL_PATH_PREFIX = '/test-root-url-prefix';
+    var res = {
+      writeHead: function () {},
+      end: function (content) {
+        __meteor_runtime_config__.ROOT_URL_PATH_PREFIX = rootUrlPathPrefix;
+        test.matches(
+          content,
+          /\/test-root-url-prefix\/packages\/oauth\/end_of_redirect_response\.js/
+        );
+      }
+    };
+    var details = {
+      credentials: {},
+      loginStyle: 'redirect',
+      query: {
+        // {"redirectUrl": "http://localhost:3000/"}
```

Comment:
Not specifying it breaks the test:
`TypeError: must start with number, buffer, array or string
    at fromObject (buffer.js:208:11)
    at new Buffer (buffer.js:65:10)
    at Object.OAuth._stateFromQuery (packages/oauth.js:99:14)
    at Object.OAuth._endOfLoginResponse (packages/oauth.js:419:25)`

Specifying it (but not http://localhost:3000/) prints a warning on the console:
`Error in OAuth Server: redirectUrl () is not on the same host as the app (http://localhost:3000/)
`

The comment I added is there simply to make clear what's in the base64 encoded string.

Would this be a better option?
```js
state: new Buffer(JSON.stringify({
  redirectUrl: __meteor_runtime_config__.ROOT_URL
}), 'binary').toString('base64')
```